src/core/classes/items-management/items-manager.ts

```diff
@@ -103,6 +103,10 @@ export abstract class CoreItemsManager<
      * @param item Item, null if none.
      */
     setSelectedItem(item: Item | null): void {
+        if (item && this.selectedItem && this.getItemId(item) === this.getItemId(this.selectedItem)) {
```

Comment:
```suggestion
        if (item === this.selectedItem) {
```

I think this is what's breaking the tests, so with this we should be ready to merge :). Although you may need to rebase because there's been a while since we looked into this. But I think it should be fine, because we haven't changed any of the related files that I'm aware.